vips/VIP-ExtensionV3.md

```diff
@@ -0,0 +1,137 @@
+---
+Title: EVM Clause Context
+Description: This VIP proposes an enhancement of the built-in extension contract to provide information about the clauses in the current transaction.
+Author: @databeforedishonor and Darren Kelly (darren.kelly@vechain.org)
+Discussions: https://vechain.discourse.group/t/mtt-induced-vulnerabilities-in-randomized-events/88
+Category:  Core
+Status: Draft
+CreatedAt: 2024-04-18
+
+---
+
+## Overview
+
+The below VIP-XXX proposal outlines an upgrade to the EVM, which will expose clause information to the smart contract, via the built-in `extension` contract
+  
+## Motivation
+
+Smart contract developers may want to access information about the current clause being executed. For example, a smart contract can limit the number of clauses in case any malicious actor could exploit the contract via multiple clauses.
```

Comment:
Please specify the malicious flow.

Reply:
added the example from discourse in gherkin, WDYT?

Reply:
https://github.com/darrenvechain/VIPs/blob/extension-v3/vips/VIP-ExtensionV3.md